tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -438,15 +470,34 @@ def enqueue_request(self,
         try:
             self.enqueue_lock.acquire()
             assert self.active, "PyExecutor has already been shutdown."
-            req_id = self.next_req_id
+            # Allocate the main request ID first
+            req_id = self._get_request_id()
+
             if self.enable_iter_perf_stats:
                 self.start_times[req_id] = time.time()
 
-            if query is not None:
-                self.request_queue.put(RequestQueueItem(req_id, request, query))
-            else:
-                self.request_queue.put(RequestQueueItem(req_id, request))
-            self.next_req_id += 1
+            # Generate child request IDs if needed
+            child_req_ids = None
```

Comment:
It seems that line480 to line 500 shares the same codes with line 343 to line 361, can we provide a helper function?